src/java.desktop/share/classes/sun/awt/image/GifImageDecoder.java

```diff
@@ -593,6 +594,16 @@ private boolean readImage(boolean first, int disposal_method, int delay)
             }
             return false;
         }
+
+        isSimpleSavedImageComparison = saved_image != null && model.equals(saved_model);
```

Comment:
Looks like we can ignore `trans_pixel >= 0 && !curframe.initialframe` check here.

I see that saved model will be null for initial frame at this point in code & if trans_pixel changes `model.equals(saved_model)` fails. So current set of checks suffice.